pkg/prometheus/promcfg.go

```diff
@@ -1094,7 +1096,9 @@ func (cg *ConfigGenerator) generateProbeConfig(
 			enforcedNamespaceLabel: cg.spec.EnforcedNamespaceLabel,
 		}
 		relabelings = append(relabelings, rcg.generate(m.Spec.Targets.Ingress.RelabelConfigs)...)
-
+		if shards > 1 {
+			relabelings = generateAddressShardingRelabelingRulesForProbes(relabelings, shards)
+		}
```

Comment:
To be consistent with how we do for service/pod monitors already.

```suggestion
		relabelings = generateAddressShardingRelabelingRulesForProbes(relabelings, shards)
```